lib/vector-core/buffers/src/disk/leveldb_buffer/reader.rs

```diff
@@ -11,6 +11,7 @@ use leveldb::database::{
 };
 use std::collections::VecDeque;
 use std::fmt::Display;
+use std::future::Future;
```

Comment:
Could / should these changes be separated out into another PR?

I see there is a bunch of stale commented code as well.

Reply:
Not having written this myself I'm concerned about pulling this work out into a separate PR. I know both the s3 changes and the disk buffer changes were validated together and if we were to separate it all out this needed change would get pushed back another day probably.

I can easily clear out the stale code though.

Reply:
Dead code removed in 0aa71d3